retrofit-converters/kotlinx-serialization/src/test/java/com/jakewharton/retrofit2/converter/kotlinx/serialization/KotlinxSerializationConverterFactoryContextualTest.kt

```diff
@@ -0,0 +1,87 @@
+package com.jakewharton.retrofit2.converter.kotlinx.serialization
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.modules.contextual
+import okhttp3.MediaType
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import retrofit2.Call
+import retrofit2.Retrofit
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.POST
+
+class KotlinxSerializationConverterFactoryContextualTest {
+  @get:Rule
+  val server = MockWebServer()
+
+  private lateinit var service: Service
+
+  interface Service {
+    @GET("/")
+    fun deserialize(): Call<User>
+
+    @POST("/")
+    fun serialize(@Body user: User): Call<Void?>
+  }
+
+  data class User(val name: String)
+
+  object UserSerializer : KSerializer<User> {
```

Comment:
Why hand-write this?

Reply:
Just to confirm it all gets wired in? That’s okay.

Reply:
Yeah I believe before we were looking up the serializer in a way that did not honor contextually-registered ones.